File: 2461/Solution.go
func maximumSubarraySum(nums []int, k int) int64 {
    mx, dup, r, prev:= 0, 0, 0, 0
    for _, n := range nums {
        mx = max(mx, n)
    }
    m := make([]int, mx + 1)
    var res int64
    var sum int64

    for ; r < k - 1; r++ {
        if m[nums[r]] > 0 {
            dup++
        }
        sum += int64(nums[r])
        m[nums[r]]++
    }

    for ; r < len(nums); r++{
        sum += int64(nums[r] - prev)
        if m[nums[r]] > 0 {
            dup++
        }

        m[nums[r]]++
        
        if dup == 0 {
            res = max(res, sum)
        }
        
        prev = nums[r - k + 1]
        if m[prev]  > 1 {
            dup--
        }
        m[prev]--
    }
    return res
}

File: 2461/Solution.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {

        int max = 0;
        for(int i : nums) max = Math.max(max, i);
        int[] m = new int[max+1];

        long res = 0;
        long sum = 0;
        int dup = 0;

        for(int i =0; i < k-1; i++){
            sum+=nums[i];
            if(m[nums[i]] >= 1){
                dup++;
            }
            m[nums[i]]++;
        }
        int prev = 0;

        for(int i = k-1; i < nums.length; i++){
            sum += nums[i] - prev;
            if(m[nums[i]] >=1) dup++;

            m[nums[i]]++;

            if(dup == 0) res = Math.max(res, sum);

            prev = nums[i - k + 1];
            if(m[prev] > 1){
                dup--;
            }
            m[prev]--;
        }

        return res;
    }
}
